### DISCUSSION TITLE
[4.0] Link picker XML field (Feature Request)

### DISCUSSION THREAD

**Discussion by pinta83** [joomla/joomla-cms#47182] [2. Cold Features]
I would much like to finally see a Link picker XML field we can use in forms.... With a behaviour like the one in JCE editor - with support for plugins that provide data for non core components.

**Comment by ghost** [comment]
@alikon can you please comment?

**Comment by brianteeman** [comment]
See https://docs.joomla.org/Menuitem_form_field_type

**Comment by pinta83** [comment]
Menuitem type is not nearly how JCE Link picker works... Menuitem is only the menu items, not extensions themselves. With JCE's link you get all core extensions (content, contacts...), and with a plugin it extends to support K2.

I've recorded a 30sec video which clearly shows - Joomla menuitem field type only has menu items, and JCE... well... all of it :)

https://www.youtube.com/watch?v=9M9Frng_HIs

**Comment by alikon** [comment]
Closing this  due to not receiving required information to determine if this is a bug or not. If you feel this still needs review, please open a new tracker entry with as much information as possible to ensure it can be reviewed properly

**Comment by brianteeman** [comment]
@alikon what is this required information you are seeking. This is a very valid feature request not a bug report.

**Comment by bembelimen** [comment]
Aren't the editor-xtd plugins exactly that?

**Comment by brianteeman** [comment]
Similar but not quite AND they're not fields

**Comment by coolcat-creations** [comment]
It would be great if Link A and Link B could get this Feature: 
select from: Menuitems, Articles, Media and External Link
